query/storage.go

```diff
@@ -0,0 +1,77 @@
+package query
```

Comment:
Go package names drive me crazy sometimes.  This is a subpackage of query, that will be referred to as `storage` inside the top-level `storage` package.

Reply:
It will be referenced as `query` in the storage package. Would you prefer it was referred to as `storage.Reader` instead?

Reply:
I renamed the type to `StorageReader` so it was clear that it was the query subsystem reading from storage rather than a generic "reader" which could have been misinterpreted as a query reader.

Reply:
Ah, right.  I was just confused.  I think the change is an improvement.